core/src/main/java/com/linecorp/armeria/common/metric/MeterRegistryUtil.java

```diff
@@ -0,0 +1,269 @@
+     * and the specified name parts.
+     */
+    public static String name(MeterRegistry registry, MeterUnit unit, MeterId id, String... name) {
+        return name(registry, unit, ImmutableList.<String>builder().add(id.getName()).add(name).build());
+    }
+
+    /**
+     * Returns a newly-created {@link Meter} name by concatenating the name of the specified {@link MeterId}
+     * and the specified name parts.
+     */
+    public static String name(MeterRegistry registry, MeterUnit unit, MeterId id, Iterable<String> name) {
+        return name(registry, unit, ImmutableList.<String>builder().add(id.getName()).addAll(name).build());
+    }
+
+    /**
+     * Returns a newly-created {@link Meter} name.
+     */
+    public static String name(MeterRegistry registry, MeterUnit unit, String... name) {
+        return name(registry, unit, ImmutableList.copyOf(name));
+    }
+
+    private static String name(MeterRegistry registry, MeterUnit unit, Iterable<String> name) {
+        if (registry instanceof PrometheusMeterRegistry) {
+            final StringBuilder buf = new StringBuilder();
+            for (String n : name) {
+                if (ALPHANUM_ONLY_PATTERN.matcher(n).matches()) {
+                    buf.append(CaseFormat.LOWER_CAMEL.to(CaseFormat.LOWER_UNDERSCORE, n));
+                } else {
+                    buf.append(n);
+                }
+                buf.append('_');
+            }
+            buf.setLength(buf.length() - 1);
+            final String unitName = unit.baseUnitName();
+            if (!unitName.isEmpty()) {
+                buf.append('_').append(unitName);
+            }
+            if (unit.isTotal()) {
+                buf.append("_total");
+            }
+
+            return PROMETHEUS_SANITIZE_BODY_PATTERN.matcher(
+                    PROMETHEUS_SANITIZE_PREFIX_PATTERN.matcher(buf.toString())
+                                                      .replaceFirst("_")).replaceAll("_");
+        }
+
+        return String.join(".", name);
+    }
+
+    /**
+     * Concatenates the {@link Tag}s of the specified {@link MeterId} and the specified {@code keyValues}.
+     */
+    public static Iterable<Tag> tags(MeterId id, String... keyValues) {
+        return tags(id.getTags(), keyValues);
+    }
+
+    /**
+     * Concatenates the specified {@link Tag}s and {@code keyValues}.
+     */
+    public static Iterable<Tag> tags(Iterable<Tag> tags, String... keyValues) {
+        return Iterables.concat(tags, Tags.zip(keyValues));
+    }
+
+    /**
+     * Returns the value of the {@link Measurement} collected from the matching {@link Meter}.
+     *
+     * @throws NoSuchElementException if there's no such {@link Meter} in the specified {@link MeterRegistry}
+     */
+    public static double measure(MeterRegistry registry, String name, String... keyValues) {
+        requireNonNull(name, "name");
```

Comment:
Seems null check for the registry is missed for all of the methods in this class here